ID problems running your script with my shiny container

Thanks for this super helpful script!

I'm trying to set up my own shiny container to work without modifications to your script, but I'm running into an ID or permissions issue.

When I run it as-is with my container, I get the following error:

```
[2018-07-17T03:12:43.488] [ERROR] shiny-server - Error loading config: The user 'rstudio' does not have permissions to run applications as one of the users in 'docker'. Please restart shiny-server as one of the users in  'docker'. (/etc/shiny-server/shiny-server.conf:1:1)
```

I've managed to get around this by changing `run_as docker;` to `run_as rstudio;` in the `shiny-server.conf` file, **after** running `prepare_template.sh start`,  and then it works fine.

I also tried adding `adduser --disabled-password --gecos "" --ingroup shiny docker` to my `$post`, but that gave me the same error.

My Singularity file is here: https://github.com/TomHarrop/singularity-containers/blob/master/tools/Singularity.shinotate_1.5.8.918

Built version here if it helps: https://www.singularity-hub.org/collections/996

I would appreciate any tips in getting my container set up properly!

--- a/README.md
+++ b/README.md
@@ -36,6 +36,7 @@ Steps:
 
           Options:  
            --port:  the port for the application (e.g., shiny default is 3737)
+           --user:  the user for the run_as directive in the shiny configuration
            --base: base folder with applications
            --logs: temporary folder with write for logs (not required)
            --disable-index: disable directory indexing
@@ -74,7 +75,7 @@ The configuration is generated in your present working directory:
 
 ```
 $ cat shiny-server.conf
-run_as docker;
+run_as vanessa;
 server {
   listen 9098;
 
@@ -147,5 +148,5 @@ the root of the server (at the port) won't be able to explore all of your apps.
 $ /bin/bash prepare_template.sh --disable-index
 ```
 
-You can also customize the port, temporary folder, and base (if somewhere other than
+You can also customize the port, temporary folder, "run_as" user, and base (if somewhere other than
 /srv/shiny-server). Have fun!

--- a/prepare_template.sh
+++ b/prepare_template.sh
@@ -17,6 +17,7 @@ usage () {
 
           Options:  
            --port:  the port for the application (e.g., shiny default is 3737)
+           --user:  the user for the run_as directive in the shiny configuration
            --base: base folder with applications
            --logs: temporary folder with write for logs (not required)
            --disable-index: disable directory indexing
@@ -43,6 +44,9 @@ SHINY_LOGS=$(mktemp -d /tmp/shiny-server.XXXXXX) && rmdir ${SHINY_LOGS};
 # Disable indexing (on, default, is not disabled)
 DISABLE_DIRINDEX="on";
 
+# User to run_as, defaults to docker
+SHINY_USER="${USER}"
+
 if [ $# -eq 0 ]; then
     usage
     exit
@@ -61,6 +65,11 @@ while true; do
         -p|--port|port)
             shift
             SHINY_PORT="${1:-}"
+            shift
+        ;;
+        -u|--user)
+            shift
+            SHINY_USER="${1:-}"
             shift
         ;;
         -di|--disable-index|disable-index)
@@ -89,7 +98,8 @@ function prepare_conf() {
     SHINY_BASE=$2
     SHINY_LOGS=$3
     DISABLE_DIRINDEX=$4
-    CONFIG="run_as docker;
+    SHINY_USER=$5
+    CONFIG="run_as ${SHINY_USER};
 server {
   listen ${SHINY_PORT};
 
@@ -119,10 +129,11 @@ if [ "${SHINY_START}" == "yes" ]; then
     echo "port: ${SHINY_PORT}";
     echo "logs:" ${SHINY_LOGS};
     echo "base: ${SHINY_BASE}";
+    echo "run_as: ${SHINY_USER}";
 
     # Prepare the template
     
-    CONFIG=$(prepare_conf $SHINY_PORT $SHINY_BASE $SHINY_LOGS $DISABLE_DIRINDEX);
+    CONFIG=$(prepare_conf $SHINY_PORT $SHINY_BASE $SHINY_LOGS $DISABLE_DIRINDEX $SHINY_USER);
     
     # Temporary directory, if doesn't exist
     if [ ! -d "${SHINY_LOGS}" ]; then

